lib/cloudregister/registerutils.py

```diff
@@ -254,8 +254,13 @@ def fetch_smt_data(cfg, proxies):
         # trying IPv6 first, if available.
         region_servers_ipv4 = []
         region_servers_ipv6 = []
+        region_servers_dns = []
         for srv in region_servers:
-            ip_addr = ipaddress.ip_address(srv.strip())
+            srv_id = srv.strip()
+            try:
+                ip_addr = ipaddress.ip_address(srv_id)
+            except ValueError:
+                region_servers_dns.append(srv_id)
             if isinstance(ip_addr, ipaddress.IPv6Address):
```

Comment:
If there are no region servers listed by IP address (only by hostname), then by the time execution reaches this line, an exception is raised because `ip_addr` is not defined.

UnboundLocalError: local variable 'ip_addr' referenced before assignment